feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/novacard/overview/webViewController/WebViewCardCreationInterceptor.kt

```diff
@@ -0,0 +1,135 @@
+package io.novafoundation.nova.feature_assets.presentation.novacard.overview.webViewController
+
+import android.webkit.CookieManager
+import android.webkit.WebResourceRequest
+import com.google.gson.Gson
+import com.google.gson.annotations.SerializedName
+import io.novafoundation.nova.common.utils.fromJson
+import io.novafoundation.nova.common.utils.inBackground
+import io.novafoundation.nova.common.utils.onEachLatest
+import io.novafoundation.nova.common.utils.readText
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.launchIn
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import kotlin.time.Duration.Companion.milliseconds
+
+interface OnCardCreatedListener {
+
+    fun onCardCreated()
+}
+
+class WebViewCardCreationInterceptorFactory(
+    private val okHttpClient: OkHttpClient,
+    private val gson: Gson
+) {
+    fun create(onCardCreatedListener: OnCardCreatedListener): WebViewCardCreationInterceptor {
+        return WebViewCardCreationInterceptor(okHttpClient, gson, onCardCreatedListener)
+    }
+}
+
+class WebViewCardCreationInterceptor(
+    private val okHttpClient: OkHttpClient,
+    private val gson: Gson,
+    private val onCardCreatedListener: OnCardCreatedListener
+) {
+
+    private var interceptedCardRequest = MutableStateFlow<Request.Builder?>(null)
+
+    fun runPolling(coroutineScope: CoroutineScope) {
+        interceptedCardRequest.onEachLatest { request ->
+            if (request == null) return@onEachLatest
+
+            for (i in 0..100) {
+                val cardWasSuccessfullyCreated = makeRequest(request)
+
+                if (cardWasSuccessfullyCreated) {
+                    break
+                }
+
+                delay(1000.milliseconds)
+            }
+        }.inBackground()
+            .launchIn(coroutineScope)
+    }
+
+    /**
+     * @return true if the request was intercepted otherwise false
+     */
+    fun intercept(request: WebResourceRequest): Boolean {
+        val url = request.url.toString()
+
+        // Intercept requests that might return a JSON response
+        if (url.contains("https://api.mercuryo.io/v1.6/cards")) { // Specify your condition here
+            performOkHttpRequest(request)
+            return true
+        }
+
+        return false
+    }
+
+    private fun performOkHttpRequest(request: WebResourceRequest) {
+        try {
+            // Create OkHttp Request based on WebResourceRequest
+            val okHttpRequestBuilder = Request.Builder().url(request.url.toString())
+
+            // Set method (GET, POST, etc.) and request body if needed
+            when (request.method) {
+                "GET" -> okHttpRequestBuilder.get()
+                else -> okHttpRequestBuilder.get()
+            }
```

Comment:
Can be simplified to just `okHttpRequestBuilder.get()` since we dont expect anything else